spmd/tensor/api.py

```diff
@@ -380,8 +381,12 @@ def _view_with_sharding_dim_change(self, sharding_dim, shape):
 
             # Infer the dim which is specified with -1.
             if infer_idx is not None:
-                st_size = math.prod(self.size())  # type: ignore[attr-defined]
-                shape_size = -1 * math.prod(shape)  # type: ignore[attr-defined]
+                st_size = reduce(
```

Comment:
Curious, why do we need to replace `math.prod` with `reduce(operator.mul, ...)`?

Reply:
Because this is only supported since Python 3.8 however the python version in the docker we used for gpu test is just 3.7...

Reply:
I see 👍🏻